packages/eas-cli/src/build/build.ts

```diff
@@ -62,9 +64,14 @@ export async function startBuildForPlatformAsync<
     });
   }
 
-  await commitChangesForBuildAsync(builder.ctx.platform, {
-    nonInteractive: builder.ctx.commandCtx.nonInteractive,
-  });
+  if (!(await isGitStatusCleanAsync())) {
+    log.addNewLineIfNone();
+    const projectType = builder.ctx.platform === Platform.Android ? 'Android' : 'Xcode';
+    await reviewAndCommitChangesAsync(`Update runtime version in the ${projectType} project`, {
+      nonInteractive: builder.ctx.commandCtx.nonInteractive,
+    });
+  }
+
```

Comment:
Shouldn't this `if` be moved inside `if (!builder.ctx.commandCtx.skipProjectConfiguration) { }`?

Reply:
it shouldn't matter, both approaches work the same. 

In terms of code structure also not necessarily, in the current implementation, only project configuration steps can modify a project, but maybe in the future, there will be sth else that does it.

Reply:
But currently, the commit message is `Update runtime version in the ${projectType} project`
IMO, this is really confusing to see something like this in function `prepareBuildRequestForPlatformAsync`.